lib/Sema/DerivedConformanceCodable.cpp

```diff
@@ -830,9 +830,17 @@ deriveBodyDecodable_init(AbstractFunctionDecl *initDecl, void *) {
       std::tie(varDecl, varType, useIfPresentVariant) =
           lookupVarDeclForCodingKeysCase(conformanceDC, elt, targetDecl);
 
-      // Don't output a decode statement for a var let with a default value.
-      if (varDecl->isLet() && varDecl->isParentInitialized())
+      // Don't output a decode statement for a let with an initial value.
+      if (varDecl->isLet() && varDecl->isParentInitialized()) {
+        // But emit a warning to let the user know that it won't be decoded
+        C.Diags.diagnose(varDecl->getLoc(),
```

Comment:
Not really much important, but do you know if this will be a place for use `Decl::diagnose`? This is really a question because I'm actually not sure :)

Reply:
I think a better way of doing this would be to use `diagnose(DeclNameLoc, ....)` method and remove name from the diagnostic itself, since we already point to the member there is no need to re-state the name.

Reply:
I can remove the property name from the warning diagnostic, but I have noticed we do mention it in many diagnostics even though it points to the decl. I think it just reads better, but I don't mind dropping it.

Do you mean `diagnose(DeclNameLoc, ....)` i.e. the free function? Because I can't access it. I can do `varDecl->diagnose(...)` or `varDecl->getDiags().diagnose(DeclNameLoc, ....)` instead though.

Reply:
You can use `varDecl->diagnose` yeah.